### PR TITLE
Accept language on 2 or 5 chars

### PR DESCRIPTION
**Pull request by j0k3r**
Accept `en`, `fr`, etc.. but also `de-DE`, `en-US`, etc.
Also replace dead image in _tests_ by a real image.
And use Regex to validate the language

Related:
- https://github.com/wallabag/wallabag/pull/5647
- https://github.com/wallabag/wallabag/issues/5596

### CODE REVIEW

File: src/PHPePub/Core/EPub.php [hunk 1]
```diff
@@ -1457,7 +1457,7 @@ function getLanguage() {
      * @return bool $success
      */
     function setLanguage($language) {
-        if ($this->isFinalized || mb_strlen($language) != 2) {
+        if ($this->isFinalized || mb_strlen($language) < 2 || mb_strlen($language) > 5) {
```
Comment (tcitworld) [inline src/PHPePub/Core/EPub.php:1460]: What about values like `zh-hant` ?

Comment (tcitworld) [reply src/PHPePub/Core/EPub.php:1460]: https://stackoverflow.com/a/17863380

Comment (Kdecherf) [reply src/PHPePub/Core/EPub.php:1460]: Also `_` is not supported, separator is `-`

Comment (tcitworld) [reply src/PHPePub/Core/EPub.php:1460]: We can validate the locale code if we require Intl https://www.php.net/manual/en/resourcebundle.locales.php

Comment (j0k3r) [reply src/PHPePub/Core/EPub.php:1460]: So lower than 2 and greater than 7 or 8 ?

Comment (j0k3r) [reply src/PHPePub/Core/EPub.php:1460]: Do we really need a complex validation for the language?

Comment (Kdecherf) [reply src/PHPePub/Core/EPub.php:1460]: You could replace mr_strlen calls with this regex:

``` php
preg_match('/^((?<language>([A-Za-z]{2,3}(-(?<extlang>[A-Za-z]{3}(-[A-Za-z]{3}){0,2}))?)|[A-Za-z]{4}|[A-Za-z]{5,8})(-(?<region>[A-Za-z]{2}|[0-9]{3}))?)$/', $language);
```

It is incomplete but should work for almost all cases.

References:
- https://github.com/sebinsua/ietf-language-tag-regex/blob/master/index.js
- https://schneegans.de/lv/#regex

Comment (Kdecherf) [reply src/PHPePub/Core/EPub.php:1460]: Note:

This regex will accept `en`, `fr`, `fr-FR`, … while refusing things like `fr-`.

**Review by Kdecherf — COMMENTED** [review]
You could replace `mb_strlen` calls with this regex:

``` php
preg_match('/^((?<language>([A-Za-z]{2,3}(-(?<extlang>[A-Za-z]{3}(-[A-Za-z]{3}){0,2}))?)|[A-Za-z]{4}|[A-Za-z]{5,8})(-(?<region>[A-Za-z]{2}|[0-9]{3}))?)$/', $language);
```

It is incomplete but should work for almost all cases.

References:
- https://github.com/sebinsua/ietf-language-tag-regex/blob/master/index.js
- https://schneegans.de/lv/#regex

**Comment by Kdecherf** [comment]
@j0k3r hm the results I see on your snippet are expected

**Comment by j0k3r** [comment]
Hmm does the comparison in my last commit is ok then?

**Review by Kdecherf — APPROVED** [review]
The regex in last commit looks good, I let you squash before merging

**Review by nicosomb — APPROVED** [review]

**Pull request merged**